src/main/java/com/karankumar/bookproject/ui/statistics/StatisticsView.java

```diff
@@ -42,13 +43,23 @@
 @PageTitle("Statistics | Book Project")
 @Log
 public class StatisticsView extends VerticalLayout {
+    private final Span errorMessage = new Span();
 
     public StatisticsView(PredefinedShelfService predefinedShelfService) {
-        for (StatisticType statistic : StatisticType.values()) {
-            String caption = statistic.getCaption();
-            Optional<String> value = statistic.calculateStatistic(predefinedShelfService);
-            value.ifPresent(val -> add(configureStatistic(caption, val)));
+        try{
+            for (StatisticType statistic : StatisticType.values()) {
+                String caption = statistic.getCaption();
+                Optional<String> value = statistic.calculateStatistic(predefinedShelfService);
+                value.ifPresent(val -> add(configureStatistic(caption, val)));
+            }
+        }catch(NullPointerException e){
+            System.out.println("NullPointerException : "+ e);
+            errorMessage.setId("error-message");
+            errorMessage.setText(
+                            "No statistics available. Try adding books you've read");
+            add(errorMessage);
         }
+        
```

Comment:
Instead of catching a null pointer exception (generally not considered good practice), could you come up with another way to show the 'no statistics' message? If you need a hand with this, feel free to reach to us on the #help channel on Slack

Reply:
ok I will try another way to catch this exception